### PR TITLE
[MODINVSTOR-1264] Add codes to Subject sources

### PR DESCRIPTION
**Pull request by RomanChernetskyi**
### Purpose
Add codes to Subject sources
### Approach

- Extend schema of Subject source to contain code (code should be unique)
- Remove reference data “specified in subfiled $2”
- Add codes to existing as ref data Subject sources (find corresponding values in [Subject Heading and Term Source Codes: Source Codes for Vocabularies, Rules, and Schemes (Network Development and MARC Standards Office, Library of Congress)](https://www.loc.gov/standards/sourcelist/subject.html) )

### Jira 
https://folio-org.atlassian.net/browse/MODINVSTOR-1264

### CODE REVIEW

**Review by psmagin — APPROVED** [review]
Please update NEWS

**Review by mukhiddin-yusuf — APPROVED** [review]

**Review by dmytrokrutii — APPROVED** [review]

File: reference-data/subject-types/Topicalterm.json
Comment (RuslanLavrov) [inline reference-data/subject-types/Topicalterm.json:1]: should we fill in this file or we don't need it?

Comment (RomanChernetskyi) [reply reference-data/subject-types/Topicalterm.json:1]: Filled file, thank you, regarding adding ref-data jsons: it is done to keep ref-data consistent and visible

**Review by KaterynaSenchenko — APPROVED** [review]

File: reference-data/subject-types/Topicalterm.json
Comment (RuslanLavrov) [reply reference-data/subject-types/Topicalterm.json:1]: thank you. However, the corresponding sql scripts already ensure consistency and visibility of source and type ref-data as well. Just thought that mixing two approaches for ref-data creation could lead to confusion about how this data is being created. Maybe we should use only one of them to maintain clarity on how the subjects source and type ref-data is created, what do you think?

Comment (RomanChernetskyi) [reply reference-data/subject-types/Topicalterm.json:1]: @psmagin confirmed that we use this approach for now to keep ref data visible

**Pull request merged**